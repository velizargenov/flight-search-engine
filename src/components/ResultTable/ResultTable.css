.result-table {
  width: 746px;
  padding: 21px 10px 0 10px;
  display: inline-block;
  overflow-y: auto;
  height: 563px;
  position: relative;
}
.result-table::-webkit-scrollbar {
  display: none;
  -ms-overflow-style: -ms-autohiding-scrollbar;
}
.result-table > h2 {
  font-size: 18px;
  font-weight: 400;
  letter-spacing: 1px;
  margin: 0;
  padding: 0;
}
.result-table > section {
  display: inline-block;
  overflow: hidden;
  max-width: 200px;
  position: absolute;
  right: 10px;
  top: 0;
  top: 21px;
}
.result-table--summary {
  display: block;
  overflow: hidden;
  max-width: 200px;
  max-height: 13px;
  font-size: 11px;
  letter-spacing: 0.7px;
  margin-bottom: 3px;
}
.result-table--summary:last-child {
  margin-bottom: 0;
}
.result-table--summary > .label {
  margin-right: 5px;
}
