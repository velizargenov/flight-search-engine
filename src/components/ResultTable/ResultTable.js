import React from 'react';
import PropTypes from 'prop-types';

import Flight from '../Flight/Flight';
import './ResultTable.css';

const ResultTable = ({ flights, isReturnFlight, departureDate, returnDate }) => (
  <div className="result-table">
    <div className="result-table--header">
      <h2>Your Results</h2>
      <section>
        <span className="result-table--summary depart">
          <span className="label">Depart: <span>{departureDate}</span></span>
        </span>
        <span className="result-table--summary return">
          {
            isReturnFlight &&
            <span className="label">Return: <span>{returnDate}</span></span>
          }
        </span>
      </section>
    </div>

    { flights.length
      ? flights.map(flight =>
          (
            <Flight
              key={flight.id}
              details={flight}
              isReturnFlight={isReturnFlight}
            />
          ))
      : <h1>No flights</h1>
    }
  </div>
);

ResultTable.defaultProps = {
  isReturnFlight: true,
  departureDate: '18-01-2018',
  returnDate: '19-01-2018',
};

ResultTable.propTypes = {
  isReturnFlight: PropTypes.bool,
  departureDate: PropTypes.string,
  returnDate: PropTypes.string,
};

export default ResultTable;
