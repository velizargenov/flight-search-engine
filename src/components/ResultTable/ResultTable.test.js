import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ResultTable from './ResultTable';

jest.mock('../../data');

describe('ResultTable Component', () => {
  const data = require('../../data');
  const wrapper = shallow(<ResultTable flights={data} />);

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should have two <div> elements', () => {
    expect(wrapper.find('div').length).toBe(2);
  });
  it('should have one <h2> element', () => {
    expect(wrapper.find('h2').length).toBe(1);
  });
  it('should match text inside <h2> element', () => {
    expect(wrapper.text()).toMatch('Your Results');
  });
  it('should have one <section> element', () => {
    expect(wrapper.find('section').length).toBe(1);
  });
  it('should have six <span> elements', () => {
    expect(wrapper.find('span').length).toBe(6);
  });
  it('should have one <h1> element', () => {
    expect(wrapper.find('h1').length).toBe(1);
  });
  it('should match text inside <h1> element', () => {
    expect(wrapper.text()).toMatch('No flights');
  });
  it('should have one wrapping <div> element', () => {
    expect(wrapper.find('div.result-table').length).toBe(1);
  });
  it('should have one <div> wrapping <h2> and <section>', () => {
    expect(wrapper.find('div.result-table--header').length).toBe(1);
  });
});
