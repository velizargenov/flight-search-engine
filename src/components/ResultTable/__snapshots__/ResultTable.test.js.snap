// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultTable Component matches snapshot 1`] = `
<div
  className="result-table"
>
  <div
    className="result-table--header"
  >
    <h2>
      Your Results
    </h2>
    <section>
      <span
        className="result-table--summary depart"
      >
        <span
          className="label"
        >
          Depart: 
          <span>
            18-01-2018
          </span>
        </span>
      </span>
      <span
        className="result-table--summary return"
      >
        <span
          className="label"
        >
          Return: 
          <span>
            19-01-2018
          </span>
        </span>
      </span>
    </section>
  </div>
  <h1>
    No flights
  </h1>
</div>
`;
